src/lib.rs

```diff
@@ -290,6 +310,19 @@ impl SignatureShare {
 #[derive(PartialEq, Eq)]
 pub struct SecretKey(Box<Fr>);
 
+impl Zeroize for SecretKey {
+    fn zeroize(&mut self) {
+        let mut repr = self.0.into_repr();
+        repr.0.zeroize();
```

Comment:
Doesn't that make a converted copy of `self.0`, and then zeroizes the copy? Does this really clear the bytes of `self.0` itself?
(Same for the polynomial types below, and with `clear_fr` itself.)

There also seems to be a `zeroize_derive` crate—but I guess that can't be applied here, because `self.0` doesn't implement `Zeroize`?

Reply:
Fixed by casting pointers